data/service/api/v1/users_datasets_create.go

```diff
@@ -69,7 +69,9 @@ func UsersDataSetsCreate(dataServiceContext dataService.Context) {
 	}
 
 	dataSet.SetUserID(&targetUserID)
-	dataSet.SetCreatedUserID(pointer.FromString(details.UserID()))
+	if details.IsUser() {
+		dataSet.SetCreatedUserID(pointer.FromString(details.UserID()))
+	}
```

Comment:
Would it be simpler, given the validation failure, to do:
```suggestion
if userID := details.UserID(); userID != "" {
    dataSet.SetCreatedUserID(pointer.FromString(userID))
}
```

Reply:
Per discussions in Slack, will leave as-is. Using the system-generated user ID for service session tokens will cause downstream issues. Plus, existing platform code that references the `UserID` from `AuthDetails` checks `IsUser` beforehand.